File: Daily Test/Java/College Events - Students 31.5.2025.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[][] e = new int[n][2];
        for(int i = 0; i < n; i++) {
            String s = sc.next();
            String t = sc.next();
            String[] a = s.split(":");
            String[] b = t.split(":");
            int st = Integer.parseInt(a[0]) * 60 + Integer.parseInt(a[1]);
            int et = Integer.parseInt(b[0]) * 60 + Integer.parseInt(b[1]);
            e[i][0] = st;
            e[i][1] = et;
        }
        Arrays.sort(e, (x, y) -> x[0] - y[0]);
        boolean ok = true;
        for(int i = 1; i < n; i++) {
            if(e[i][0] < e[i-1][1]) {
                ok = false;
                break;
            }
        }
        System.out.println(ok ? "YES" : "NO");
    }
}

File: Daily Test/Python/College Events - Students 31.5.2025.py
n = int(input())
e = []

for _ in range(n):
    s, t = input().split()
    h1, m1 = map(int, s.split(":"))
    h2, m2 = map(int, t.split(":"))
    sm = h1 * 60 + m1
    em = h2 * 60 + m2
    e.append((sm, em))

e.sort()
f = 1
for i in range(1, n):
    if e[i][0] < e[i-1][1]:
        f = 0
        break

print("YES" if f else "NO")